test/integration/apiserver/admissionwebhook/match_conditions_test.go

```diff
@@ -346,8 +633,8 @@ func TestMatchConditions(t *testing.T) {
 	for _, testcase := range testcases {
 		t.Run(testcase.name, func(t *testing.T) {
 			upCh := recorder.Reset()
-
-			server, err := apiservertesting.StartTestServer(t, nil, []string{
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.StrictCostEnforcementForWebhooks, false)
```

Comment:
@jpbetz So the issue here is that the feature gate is set before emulation version is set. I think a naive fix is just to add the emulation version check before setting the feature gate?

```
+featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.31"))
featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.StrictCostEnforcementForWebhooks, false)
```

Reply:
I can live with requiring emulation version be set before feature gates during test.

If these operations are done in the wrong order it would be bad to allow the feature gates to get into a weird state, so I suspect it would be best have `SetFeatureGateEmulationVersionDuringTest` fail with a clear error.

Reply:
Ok @cici37 you can disregard my PR earlier and just explicitly set the emulation version before disabling the gate.

Reply:
The test passed after set reset the version. I'll wait for https://github.com/kubernetes/kubernetes/pull/127643 to merge first.. Thank you